### ISSUE TITLE
Make it possible to read files from other partitions and samba shares.

### ISSUE THREAD

**Issue by ThanosApostolou** [flathub/org.clementine_player.Clementine#16]
I have half of my music at a separate partition and the other half at my home server shared with samba. By default this app cannot see any of my files.

I have done `flatpak override org.clementine_player.Clementine --filesystem=host` so that the app sees my mounted partition. I don't know how to make it see my samba share which is using gvfs/gio (for example flatpak keepasxc can see it properly but I cannot figure out the needed permissions).

Please consider fixing these by default, I don't think that in 2019 anyone is actually storing his music at xdg-music folder...

**Comment by FakeShemp** [comment]
I think that's actually a Clementine/Qt problem. As I understand it, if Clementine uses xdg-portals as it should, smb should just work without anything special needed to be done.

Regarding xdg-music, I don't think anyone in 2019 is actually storing music at all. Welcome to streaming ;) But yeah, it makes no sense to by default give a music player access to the host system. xdg-music and portals exist for a reason, and you're still free to override these at your leisure.

**Comment by ThanosApostolou** [comment]
Well I disagree.... I believe that every app which reads or writes files from a user given location should have access at least at home and media (since there isn't a separate filesystem=media permission as far as I know, then that means filesystem=host for now). However, this seems to be a common policy at every flathub app. I really tried and switched about 15 apps from fedora repositories to flathub for about a week and I concluded that most of them are not fully usable out of the box. So I'll never be able to suggest them to my friends and family that I have switched over to linux. Therefore I stop using flathub myself and I switch back to my distro's repositories apps (for example clementine from repositories sees my samba share just fine).

Since I won't be able to test this bug for any updates feel free to close it if you want (or maybe keep it open in case some else wants to give input).

**Comment by FakeShemp** [comment]
Well, if you don't want sandboxed apps it makes sense to not use or recommend flatpaks, as that's what they are.

**Comment by ThanosApostolou** [comment]
Well even in android when all apps are sandboxed by design, all music players I've tried can access my phone's files and the sdcard. It makes no sense to me that most flathub apps cannot see files in home and media (partitions, sdcards, usb-drives). I don't like filesystem=host either, but until there are more permission options and better usage of portals, usability should be preferred over sandboxing.

Also flatpak (and flathub) could benefit developers and users much more than just sandboxing because of the ability to test them against one runtime and the modularity with the ability to have specific libraries versions without messing with the system's libraries. So I would prefer to use flathub apps over my distribution's (especially for my familty which most of them are limited to really old software from debian's repositories) even if they didn't offer sandboxing at all. I just cannot use them (or suggest others to use them) with current excessive sandboxing defaults.

**Comment by FakeShemp** [comment]
But android asks you if you "want to give the app access to x". By default you don't have access. And you never have host access no matter what you do. 

Why do the access have to be by default? Why can't you just allow the app the access yourself, like on android.

**Comment by ThanosApostolou** [comment]
Because there's no way currently of automatically asking for permissions. Also there is no way of changing permissions without terminal if you don't use Gnome (none of my family and friends do). So when someone opens clementine then by default he won't be able to see any music and no permissions would be asked. Another example if my mother tries to open her geogebra files from the flathub app then she won't be able to do it without knowing why. So I believe the equivalent of the asked permissions in a new android application is the default permissions in a flathub application. Conclusion, in its current state flathub is just not usable for the majority users, and painful even for the experienced users if they have more than 20 flathub apps and need to use them in their full potentials.

**Comment by FakeShemp** [comment]
But that isn't in the slightest an issue with this flatpak app. Whatever gui you use to install your flatpak apps should add support for changing permissions and then you can.

**Comment by ThanosApostolou** [comment]
Yeah ok I see your point, that the gui should handle that and not give permissions by default. I personally would prefer it if there were more relaxed permissions by default (since it would still be more contained than appimages and official/unofficial repositories), but I cannot go against your philosophy, so I guess the only hope is to have more advanced guis for managing permissions in the future. Anyway, thx for your time responding and explaining your points in details.

**Comment by FakeShemp** [comment]
All discussion within Flathub goes the opposite direction; apps need to be a lot stricter on permissions and then the underlying system of flatpak, portals, bubblewrap etc. should do all the heavy lifting communicating with the system in the safest way possible. And a gui should help the user if they like to do anything above that, but as you say, only gnome seem to have implemented anything like that so far.

I've been testing this a lot now and portals seem to be working like they should. smb shares are visible to Clementine *if* they are mounted on the host system. I've been digging into the code and as far as I can tell, Clementine doesn't have any native smb support itself (i.e. you can't mount an smb partition from within Clementine) but anything mounted by the system is accessible to me via portals.

Regarding `/media` I think I've found one solution, to add `--filesystem=/media` and let the app talk to the `UDisks2` dbus, but I have no idea if it actually works in practice. It still pokes holes in the sandboxing, but it's better than `host` access. It's tested in #21

**Comment by Justinzobel** [comment]
Now that PR 21 has been merged can this be closed?

**Comment by Justinzobel** [comment]
I'm going to close this. If you are still unable to access files on your server or mounted drives, please comment so we can investigate further and reopen this.
